src/str.rs

```diff
@@ -0,0 +1,12 @@
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Str(*const str);
```

Comment:
Hi Marco! PR looks pretty good 👌 

My only question would be: why not just directly put the `*const str` into `Value::String` and skip this wrapper struct? Should be possible to directly allocate the `str` into our bump-allocated heap?